{"cells":[{"cell_type":"markdown","metadata":{"id":"VY5ilTHy_WxF"},"source":["# Setup"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"tv7US0CF_HyE"},"outputs":[],"source":["%reload_ext autoreload\n","%autoreload 2\n","\n","from IPython.core.interactiveshell import InteractiveShell\n","InteractiveShell.ast_node_interactivity = \"all\""]},{"cell_type":"code","execution_count":null,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"5ing0udi_eZe","executionInfo":{"status":"ok","timestamp":1742172470556,"user_tz":-120,"elapsed":25988,"user":{"displayName":"Ni Ma","userId":"02197765216246653270"}},"outputId":"86ad6b66-7b06-4c53-b540-10588efab207"},"outputs":[{"output_type":"stream","name":"stdout","text":["Mounted at /content/gdrive\n"]}],"source":["from google.colab import drive\n","drive.mount('/content/gdrive', force_remount = True)"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"A7Xaez3cAtHH"},"outputs":[],"source":["SYS_PROJECT_DIR  = '/content/gdrive/MyDrive/Colab Notebooks/ocr'\n","SYS_SRC_DIR  = '/content/gdrive/MyDrive/Colab Notebooks/ocr/ocr'\n","# SYS_CONFIG_DIR  = '/content/gdrive/MyDrive/Colab Notebooks/deoptima/deoptima/deoptima/config'\n","# SYS_RUNTIME_DIR = '/content/gdrive/MyDrive/Colab Notebooks/deoptima/deoptima/runtime'"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"SvDFm5dAAih0"},"outputs":[],"source":["import os\n","import pandas as pd\n","import shutil\n","\n","pd.set_option('display.max_columns', None)\n","pd.set_option('display.width', None)\n","pd.set_option('display.max_colwidth', None)\n","pd.options.mode.copy_on_write = True"]},{"cell_type":"markdown","source":["# Download"],"metadata":{"id":"aVhbXP-wt_8f"}},{"cell_type":"code","source":["# SYS_DOWNLOAD_DIR  = '/content/gdrive/MyDrive/Colab Notebooks/ocr/ocr-benchmark'"],"metadata":{"id":"m8_064mMzK3s"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["# !git clone https://github.com/video-db/ocr-benchmark.git \"{SYS_DOWNLOAD_DIR}\""],"metadata":{"id":"u09mMe4muCYj"},"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"kMO4ESpI5F6M"},"source":["# Installs"]},{"cell_type":"code","execution_count":null,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"GmX0Qdub2tni","executionInfo":{"status":"ok","timestamp":1742172471721,"user_tz":-120,"elapsed":714,"user":{"displayName":"Ni Ma","userId":"02197765216246653270"}},"outputId":"1229cc61-0d02-4d2a-d33b-3e9dd6ed0382"},"outputs":[{"output_type":"execute_result","data":{"text/plain":["269"]},"metadata":{},"execution_count":7}],"source":["import os\n","\n","requirements = \"\"\"\n","python-dotenv==1.0.1\n","imageio==2.37.0\n","numpy==2.2.2\n","onnxruntime==1.20.1\n","openai==1.60.2\n","opencv-python==4.11.0.86\n","opencv-python-headless==4.11.0.86\n","pillow==11.1.0\n","python-Levenshtein==0.26.1\n","PyYAML==6.0.2\n","videodb==0.2.10\n","grpcio==1.60.1\n","mistralai==1.5.1\n","google-genai==1.4.0\n","\"\"\"\n","\n","requirements_path = os.path.join(SYS_SRC_DIR, 'requirements.txt')\n","with open(requirements_path, 'w') as f:\n","    f.write(requirements)"]},{"cell_type":"code","source":["!pip install google-genai"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"s2zYFfPgcVTX","executionInfo":{"status":"ok","timestamp":1742172474652,"user_tz":-120,"elapsed":2930,"user":{"displayName":"Ni Ma","userId":"02197765216246653270"}},"outputId":"4735a349-1be8-4be9-ac0e-849c71afc08b"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Requirement already satisfied: google-genai in /usr/local/lib/python3.11/dist-packages (1.4.0)\n","Requirement already satisfied: google-auth<3.0.0dev,>=2.14.1 in /usr/local/lib/python3.11/dist-packages (from google-genai) (2.38.0)\n","Requirement already satisfied: httpx<1.0.0dev,>=0.28.1 in /usr/local/lib/python3.11/dist-packages (from google-genai) (0.28.1)\n","Requirement already satisfied: pydantic<3.0.0dev,>=2.0.0 in /usr/local/lib/python3.11/dist-packages (from google-genai) (2.10.6)\n","Requirement already satisfied: requests<3.0.0dev,>=2.28.1 in /usr/local/lib/python3.11/dist-packages (from google-genai) (2.32.3)\n","Requirement already satisfied: websockets<15.0dev,>=13.0 in /usr/local/lib/python3.11/dist-packages (from google-genai) (14.2)\n","Requirement already satisfied: typing-extensions<5.0.0dev,>=4.11.0 in /usr/local/lib/python3.11/dist-packages (from google-genai) (4.12.2)\n","Requirement already satisfied: cachetools<6.0,>=2.0.0 in /usr/local/lib/python3.11/dist-packages (from google-auth<3.0.0dev,>=2.14.1->google-genai) (5.5.2)\n","Requirement already satisfied: pyasn1-modules>=0.2.1 in /usr/local/lib/python3.11/dist-packages (from google-auth<3.0.0dev,>=2.14.1->google-genai) (0.4.1)\n","Requirement already satisfied: rsa<5,>=3.1.4 in /usr/local/lib/python3.11/dist-packages (from google-auth<3.0.0dev,>=2.14.1->google-genai) (4.9)\n","Requirement already satisfied: anyio in /usr/local/lib/python3.11/dist-packages (from httpx<1.0.0dev,>=0.28.1->google-genai) (3.7.1)\n","Requirement already satisfied: certifi in /usr/local/lib/python3.11/dist-packages (from httpx<1.0.0dev,>=0.28.1->google-genai) (2025.1.31)\n","Requirement already satisfied: httpcore==1.* in /usr/local/lib/python3.11/dist-packages (from httpx<1.0.0dev,>=0.28.1->google-genai) (1.0.7)\n","Requirement already satisfied: idna in /usr/local/lib/python3.11/dist-packages (from httpx<1.0.0dev,>=0.28.1->google-genai) (3.10)\n","Requirement already satisfied: h11<0.15,>=0.13 in /usr/local/lib/python3.11/dist-packages (from httpcore==1.*->httpx<1.0.0dev,>=0.28.1->google-genai) (0.14.0)\n","Requirement already satisfied: annotated-types>=0.6.0 in /usr/local/lib/python3.11/dist-packages (from pydantic<3.0.0dev,>=2.0.0->google-genai) (0.7.0)\n","Requirement already satisfied: pydantic-core==2.27.2 in /usr/local/lib/python3.11/dist-packages (from pydantic<3.0.0dev,>=2.0.0->google-genai) (2.27.2)\n","Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.11/dist-packages (from requests<3.0.0dev,>=2.28.1->google-genai) (3.4.1)\n","Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.11/dist-packages (from requests<3.0.0dev,>=2.28.1->google-genai) (2.3.0)\n","Requirement already satisfied: pyasn1<0.7.0,>=0.4.6 in /usr/local/lib/python3.11/dist-packages (from pyasn1-modules>=0.2.1->google-auth<3.0.0dev,>=2.14.1->google-genai) (0.6.1)\n","Requirement already satisfied: sniffio>=1.1 in /usr/local/lib/python3.11/dist-packages (from anyio->httpx<1.0.0dev,>=0.28.1->google-genai) (1.3.1)\n"]}]},{"cell_type":"code","execution_count":null,"metadata":{"id":"DjyetbztnSlJ"},"outputs":[],"source":["# import locale\n","# locale.getpreferredencoding = lambda: \"UTF-8\""]},{"cell_type":"code","execution_count":null,"metadata":{"id":"JIMNva_V5IrS","executionInfo":{"status":"ok","timestamp":1742172496574,"user_tz":-120,"elapsed":21888,"user":{"displayName":"Ni Ma","userId":"02197765216246653270"}},"colab":{"base_uri":"https://localhost:8080/"},"outputId":"957448b2-8658-4d37-f349-d57ac9d31b4f"},"outputs":[{"output_type":"stream","name":"stdout","text":["\u001b[?25l     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m0.0/62.0 kB\u001b[0m \u001b[31m?\u001b[0m eta \u001b[36m-:--:--\u001b[0m\r\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m62.0/62.0 kB\u001b[0m \u001b[31m2.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m16.4/16.4 MB\u001b[0m \u001b[31m68.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m13.3/13.3 MB\u001b[0m \u001b[31m58.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m456.1/456.1 kB\u001b[0m \u001b[31m27.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m5.4/5.4 MB\u001b[0m \u001b[31m79.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m278.3/278.3 kB\u001b[0m \u001b[31m17.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m162.7/162.7 kB\u001b[0m \u001b[31m10.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m46.0/46.0 kB\u001b[0m \u001b[31m2.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m86.8/86.8 kB\u001b[0m \u001b[31m5.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m3.1/3.1 MB\u001b[0m \u001b[31m74.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25h  Building wheel for videodb (setup.py) ... \u001b[?25l\u001b[?25hdone\n","\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n","numba 0.60.0 requires numpy<2.1,>=1.22, but you have numpy 2.2.2 which is incompatible.\n","tensorflow 2.18.0 requires numpy<2.1.0,>=1.26.0, but you have numpy 2.2.2 which is incompatible.\n","thinc 8.2.5 requires numpy<2.0.0,>=1.19.0; python_version >= \"3.9\", but you have numpy 2.2.2 which is incompatible.\n","grpcio-status 1.62.3 requires grpcio>=1.62.3, but you have grpcio 1.60.1 which is incompatible.\n","pytensor 2.27.1 requires numpy<2,>=1.17.0, but you have numpy 2.2.2 which is incompatible.\u001b[0m\u001b[31m\n","\u001b[0m"]}],"source":["import os\n","requirements_path = os.path.join(SYS_SRC_DIR, 'requirements.txt')\n","!pip install -q -r '{requirements_path}'"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"gr_5DHGsMvOV","executionInfo":{"status":"ok","timestamp":1742172497080,"user_tz":-120,"elapsed":505,"user":{"displayName":"Ni Ma","userId":"02197765216246653270"}},"colab":{"base_uri":"https://localhost:8080/"},"outputId":"e3b37439-fb42-4048-da16-4133334decc7"},"outputs":[{"output_type":"stream","name":"stderr","text":["[autoreload of numpy.core.multiarray failed: Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","             ^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/imp.py\", line 315, in reload\n","    return importlib.reload(module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/importlib/__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 621, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/core/multiarray.py\", line 1, in <module>\n","    from numpy._core import multiarray\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/_core/multiarray.py\", line 97, in <module>\n","    _override___module__()\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/_core/multiarray.py\", line 93, in _override___module__\n","    ufunc.__module__ = \"numpy\"\n","    ^^^^^^^^^^^^^^^^\n","AttributeError: 'numpy.ufunc' object has no attribute '__module__'\n","]\n","[autoreload of numpy.core.fromnumeric failed: Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 374, in superreload\n","    if not hasattr(obj, '__module__') or obj.__module__ != module.__name__:\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/core/umath.py\", line 2, in __getattr__\n","    from numpy._core import umath\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/_core/umath.py\", line 15, in <module>\n","    from ._multiarray_umath import (\n","ImportError: cannot import name '_get_extobj_dict' from 'numpy.core._multiarray_umath' (/usr/local/lib/python3.11/dist-packages/numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so)\n","]\n","[autoreload of numpy.core.numeric failed: Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 374, in superreload\n","    if not hasattr(obj, '__module__') or obj.__module__ != module.__name__:\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/core/umath.py\", line 2, in __getattr__\n","    from numpy._core import umath\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/_core/umath.py\", line 15, in <module>\n","    from ._multiarray_umath import (\n","ImportError: cannot import name '_get_extobj_dict' from 'numpy.core._multiarray_umath' (/usr/local/lib/python3.11/dist-packages/numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so)\n","]\n","[autoreload of numpy.core._internal failed: Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","             ^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/imp.py\", line 315, in reload\n","    return importlib.reload(module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/importlib/__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 621, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/core/_internal.py\", line 1, in <module>\n","    from numpy._core import _internal\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/_core/_internal.py\", line 14, in <module>\n","    from .multiarray import dtype, array, ndarray, promote_types, StringDType\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/_core/multiarray.py\", line 97, in <module>\n","    _override___module__()\n","AttributeError: 'numpy.ufunc' object has no attribute '__module__'\n","]\n","[autoreload of numpy.core failed: Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 374, in superreload\n","    if not hasattr(obj, '__module__') or obj.__module__ != module.__name__:\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/core/umath.py\", line 2, in __getattr__\n","    from numpy._core import umath\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/_core/umath.py\", line 15, in <module>\n","    from ._multiarray_umath import (\n","ImportError: cannot import name '_get_extobj_dict' from 'numpy.core._multiarray_umath' (/usr/local/lib/python3.11/dist-packages/numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so)\n","]\n","[autoreload of numpy.lib.mixins failed: Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 374, in superreload\n","    if not hasattr(obj, '__module__') or obj.__module__ != module.__name__:\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/core/umath.py\", line 2, in __getattr__\n","    from numpy._core import umath\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/_core/umath.py\", line 15, in <module>\n","    from ._multiarray_umath import (\n","ImportError: cannot import name '_get_extobj_dict' from 'numpy.core._multiarray_umath' (/usr/local/lib/python3.11/dist-packages/numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so)\n","]\n","[autoreload of numpy.lib.scimath failed: Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","             ^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/imp.py\", line 315, in reload\n","    return importlib.reload(module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/importlib/__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 621, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/lib/scimath.py\", line 1, in <module>\n","    from ._scimath_impl import (\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/lib/_scimath_impl.py\", line 17, in <module>\n","    import numpy._core.numeric as nx\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/_core/numeric.py\", line 11, in <module>\n","    from . import multiarray\n","AttributeError: 'numpy.ufunc' object has no attribute '__module__'\n","]\n","[autoreload of numpy.lib.stride_tricks failed: Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","             ^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/imp.py\", line 315, in reload\n","    return importlib.reload(module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/importlib/__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 621, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/lib/stride_tricks.py\", line 1, in <module>\n","    from ._stride_tricks_impl import (\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/lib/_stride_tricks_impl.py\", line 8, in <module>\n","    from numpy._core.numeric import normalize_axis_tuple\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/_core/numeric.py\", line 11, in <module>\n","    from . import multiarray\n","AttributeError: 'numpy.ufunc' object has no attribute '__module__'\n","]\n","/usr/local/lib/python3.11/dist-packages/numpy/_typing/_scalars.py:12: FutureWarning: In the future `np.bool` will be defined as the corresponding NumPy scalar.\n","  _BoolLike_co: TypeAlias = bool | np.bool\n","[autoreload of numpy._typing._scalars failed: Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","             ^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/imp.py\", line 315, in reload\n","    return importlib.reload(module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/importlib/__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 621, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/_typing/_scalars.py\", line 12, in <module>\n","    _BoolLike_co: TypeAlias = bool | np.bool\n","                                     ^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/__init__.py\", line 324, in __getattr__\n","    warnings.filterwarnings(\"ignore\", message=\"numpy.ndarray size changed\")\n","            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","AttributeError: module 'numpy' has no attribute 'bool'.\n","`np.bool` was a deprecated alias for the builtin `bool`. To avoid this error in existing code, use `bool` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.bool_` here.\n","The aliases was originally deprecated in NumPy 1.20; for more details and guidance see the original release note at:\n","    https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","]\n","/usr/local/lib/python3.11/dist-packages/numpy/_typing/_dtype_like.py:141: FutureWarning: In the future `np.bool` will be defined as the corresponding NumPy scalar.\n","  | type[np.bool]\n","[autoreload of numpy._typing._dtype_like failed: Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","             ^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/imp.py\", line 315, in reload\n","    return importlib.reload(module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/importlib/__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 621, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/_typing/_dtype_like.py\", line 141, in <module>\n","    | type[np.bool]\n","           ^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/__init__.py\", line 324, in __getattr__\n","    warnings.filterwarnings(\"ignore\", message=\"numpy.ndarray size changed\")\n","            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","AttributeError: module 'numpy' has no attribute 'bool'.\n","`np.bool` was a deprecated alias for the builtin `bool`. To avoid this error in existing code, use `bool` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.bool_` here.\n","The aliases was originally deprecated in NumPy 1.20; for more details and guidance see the original release note at:\n","    https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n","]\n","[autoreload of numpy._typing._array_like failed: Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","             ^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/imp.py\", line 315, in reload\n","    return importlib.reload(module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/importlib/__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 621, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/_typing/_array_like.py\", line 33, in <module>\n","    from numpy._core.multiarray import StringDType\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/_core/multiarray.py\", line 97, in <module>\n","    _override___module__()\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/_core/multiarray.py\", line 93, in _override___module__\n","    ufunc.__module__ = \"numpy\"\n","    ^^^^^^^^^^^^^^^^\n","AttributeError: 'numpy.ufunc' object has no attribute '__module__'\n","]\n","[autoreload of numpy._typing failed: Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","             ^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/imp.py\", line 315, in reload\n","    return importlib.reload(module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/importlib/__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 621, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/_typing/__init__.py\", line 120, in <module>\n","    from ._array_like import (\n","ImportError: cannot import name '_ArrayLikeFloat64_co' from 'numpy._typing._array_like' (/usr/local/lib/python3.11/dist-packages/numpy/_typing/_array_like.py)\n","]\n","[autoreload of numpy.linalg failed: Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 374, in superreload\n","    if not hasattr(obj, '__module__') or obj.__module__ != module.__name__:\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/linalg/linalg.py\", line 3, in __getattr__\n","    from numpy.linalg import _linalg\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/linalg/_linalg.py\", line 25, in <module>\n","    from numpy._core import (\n","ImportError: cannot import name 'array' from 'numpy._core' (/usr/local/lib/python3.11/dist-packages/numpy/_core/__init__.py)\n","]\n","[autoreload of numpy.matrixlib.defmatrix failed: Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","             ^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/imp.py\", line 315, in reload\n","    return importlib.reload(module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/importlib/__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 621, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/matrixlib/defmatrix.py\", line 8, in <module>\n","    import numpy._core.numeric as N\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/_core/numeric.py\", line 11, in <module>\n","    from . import multiarray\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/_core/multiarray.py\", line 97, in <module>\n","    _override___module__()\n","AttributeError: 'numpy.ufunc' object has no attribute '__module__'\n","]\n","[autoreload of numpy.lib.format failed: Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","             ^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/imp.py\", line 315, in reload\n","    return importlib.reload(module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/importlib/__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 621, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/lib/format.py\", line 170, in <module>\n","    from numpy.lib._utils_impl import drop_metadata\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/lib/_utils_impl.py\", line 9, in <module>\n","    from numpy._core import ndarray\n","ImportError: cannot import name 'ndarray' from 'numpy._core' (/usr/local/lib/python3.11/dist-packages/numpy/_core/__init__.py)\n","]\n","[autoreload of numpy.lib._iotools failed: Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","             ^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/imp.py\", line 315, in reload\n","    return importlib.reload(module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/importlib/__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 621, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/lib/_iotools.py\", line 7, in <module>\n","    import numpy._core.numeric as nx\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/_core/numeric.py\", line 11, in <module>\n","    from . import multiarray\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/_core/multiarray.py\", line 97, in <module>\n","    _override___module__()\n","AttributeError: 'numpy.ufunc' object has no attribute '__module__'\n","]\n","[autoreload of numpy.lib.npyio failed: Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","             ^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/imp.py\", line 315, in reload\n","    return importlib.reload(module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/importlib/__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 621, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/lib/npyio.py\", line 1, in <module>\n","    from ._npyio_impl import (\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/lib/_npyio_impl.py\", line 20, in <module>\n","    from numpy._core.multiarray import packbits, unpackbits\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/_core/multiarray.py\", line 97, in <module>\n","    _override___module__()\n","AttributeError: 'numpy.ufunc' object has no attribute '__module__'\n","]\n","[autoreload of numpy.lib failed: Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","             ^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/imp.py\", line 315, in reload\n","    return importlib.reload(module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/importlib/__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 621, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/lib/__init__.py\", line 13, in <module>\n","    from . import array_utils\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/lib/array_utils.py\", line 1, in <module>\n","    from ._array_utils_impl import (\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/lib/_array_utils_impl.py\", line 4, in <module>\n","    from numpy._core import asarray\n","ImportError: cannot import name 'asarray' from 'numpy._core' (/usr/local/lib/python3.11/dist-packages/numpy/_core/__init__.py)\n","]\n","[autoreload of numpy.fft._pocketfft failed: Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","             ^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/imp.py\", line 315, in reload\n","    return importlib.reload(module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/importlib/__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 621, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/fft/_pocketfft.py\", line 36, in <module>\n","    from numpy.lib.array_utils import normalize_axis_index\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/lib/array_utils.py\", line 1, in <module>\n","    from ._array_utils_impl import (\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/lib/_array_utils_impl.py\", line 4, in <module>\n","    from numpy._core import asarray\n","ImportError: cannot import name 'asarray' from 'numpy._core' (/usr/local/lib/python3.11/dist-packages/numpy/_core/__init__.py)\n","]\n","[autoreload of numpy.fft failed: Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 374, in superreload\n","    if not hasattr(obj, '__module__') or obj.__module__ != module.__name__:\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/fft/helper.py\", line 3, in __getattr__\n","    from numpy.fft import _helper\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/fft/_helper.py\", line 5, in <module>\n","    from numpy._core import integer, empty, arange, asarray, roll\n","ImportError: cannot import name 'integer' from 'numpy._core' (/usr/local/lib/python3.11/dist-packages/numpy/_core/__init__.py)\n","]\n","[autoreload of numpy.polynomial.polyutils failed: Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","             ^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/imp.py\", line 315, in reload\n","    return importlib.reload(module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/importlib/__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 621, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/polynomial/polyutils.py\", line 27, in <module>\n","    from numpy._core.multiarray import dragon4_positional, dragon4_scientific\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/_core/multiarray.py\", line 97, in <module>\n","    _override___module__()\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/_core/multiarray.py\", line 93, in _override___module__\n","    ufunc.__module__ = \"numpy\"\n","    ^^^^^^^^^^^^^^^^\n","AttributeError: 'numpy.ufunc' object has no attribute '__module__'\n","]\n","[autoreload of numpy.polynomial.polynomial failed: Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","             ^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/imp.py\", line 315, in reload\n","    return importlib.reload(module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/importlib/__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 621, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/polynomial/polynomial.py\", line 85, in <module>\n","    from numpy.lib.array_utils import normalize_axis_index\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/lib/array_utils.py\", line 1, in <module>\n","    from ._array_utils_impl import (\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/lib/_array_utils_impl.py\", line 4, in <module>\n","    from numpy._core import asarray\n","ImportError: cannot import name 'asarray' from 'numpy._core' (/usr/local/lib/python3.11/dist-packages/numpy/_core/__init__.py)\n","]\n","[autoreload of numpy.polynomial.chebyshev failed: Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","             ^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/imp.py\", line 315, in reload\n","    return importlib.reload(module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/importlib/__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 621, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/polynomial/chebyshev.py\", line 112, in <module>\n","    from numpy.lib.array_utils import normalize_axis_index\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/lib/array_utils.py\", line 1, in <module>\n","    from ._array_utils_impl import (\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/lib/_array_utils_impl.py\", line 4, in <module>\n","    from numpy._core import asarray\n","ImportError: cannot import name 'asarray' from 'numpy._core' (/usr/local/lib/python3.11/dist-packages/numpy/_core/__init__.py)\n","]\n","[autoreload of numpy.polynomial.legendre failed: Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","             ^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/imp.py\", line 315, in reload\n","    return importlib.reload(module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/importlib/__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 621, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/polynomial/legendre.py\", line 84, in <module>\n","    from numpy.lib.array_utils import normalize_axis_index\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/lib/array_utils.py\", line 1, in <module>\n","    from ._array_utils_impl import (\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/lib/_array_utils_impl.py\", line 4, in <module>\n","    from numpy._core import asarray\n","ImportError: cannot import name 'asarray' from 'numpy._core' (/usr/local/lib/python3.11/dist-packages/numpy/_core/__init__.py)\n","]\n","[autoreload of numpy.polynomial.hermite failed: Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","             ^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/imp.py\", line 315, in reload\n","    return importlib.reload(module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/importlib/__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 621, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/polynomial/hermite.py\", line 80, in <module>\n","    from numpy.lib.array_utils import normalize_axis_index\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/lib/array_utils.py\", line 1, in <module>\n","    from ._array_utils_impl import (\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/lib/_array_utils_impl.py\", line 4, in <module>\n","    from numpy._core import asarray\n","ImportError: cannot import name 'asarray' from 'numpy._core' (/usr/local/lib/python3.11/dist-packages/numpy/_core/__init__.py)\n","]\n","[autoreload of numpy.polynomial.hermite_e failed: Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","             ^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/imp.py\", line 315, in reload\n","    return importlib.reload(module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/importlib/__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 621, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/polynomial/hermite_e.py\", line 80, in <module>\n","    from numpy.lib.array_utils import normalize_axis_index\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/lib/array_utils.py\", line 1, in <module>\n","    from ._array_utils_impl import (\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/lib/_array_utils_impl.py\", line 4, in <module>\n","    from numpy._core import asarray\n","ImportError: cannot import name 'asarray' from 'numpy._core' (/usr/local/lib/python3.11/dist-packages/numpy/_core/__init__.py)\n","]\n","[autoreload of numpy.polynomial.laguerre failed: Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","             ^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/imp.py\", line 315, in reload\n","    return importlib.reload(module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/importlib/__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 621, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/polynomial/laguerre.py\", line 80, in <module>\n","    from numpy.lib.array_utils import normalize_axis_index\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/lib/array_utils.py\", line 1, in <module>\n","    from ._array_utils_impl import (\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/lib/_array_utils_impl.py\", line 4, in <module>\n","    from numpy._core import asarray\n","ImportError: cannot import name 'asarray' from 'numpy._core' (/usr/local/lib/python3.11/dist-packages/numpy/_core/__init__.py)\n","]\n","[autoreload of numpy.ctypeslib failed: Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","             ^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/imp.py\", line 315, in reload\n","    return importlib.reload(module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/importlib/__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 621, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/ctypeslib.py\", line 57, in <module>\n","    from numpy._core.multiarray import _flagdict, flagsobj\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/_core/multiarray.py\", line 97, in <module>\n","    _override___module__()\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/_core/multiarray.py\", line 93, in _override___module__\n","    ufunc.__module__ = \"numpy\"\n","    ^^^^^^^^^^^^^^^^\n","AttributeError: 'numpy.ufunc' object has no attribute '__module__'\n","]\n","[autoreload of numpy.ma.core failed: Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 374, in superreload\n","    if not hasattr(obj, '__module__') or obj.__module__ != module.__name__:\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/core/umath.py\", line 2, in __getattr__\n","    from numpy._core import umath\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/_core/umath.py\", line 15, in <module>\n","    from ._multiarray_umath import (\n","ImportError: cannot import name '_get_extobj_dict' from 'numpy.core._multiarray_umath' (/usr/local/lib/python3.11/dist-packages/numpy/core/_multiarray_umath.cpython-311-x86_64-linux-gnu.so)\n","]\n","[autoreload of numpy.ma.extras failed: Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","             ^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/imp.py\", line 315, in reload\n","    return importlib.reload(module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/importlib/__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 621, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/ma/extras.py\", line 35, in <module>\n","    from numpy.lib.array_utils import normalize_axis_index, normalize_axis_tuple\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/lib/array_utils.py\", line 1, in <module>\n","    from ._array_utils_impl import (\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/lib/_array_utils_impl.py\", line 4, in <module>\n","    from numpy._core import asarray\n","ImportError: cannot import name 'asarray' from 'numpy._core' (/usr/local/lib/python3.11/dist-packages/numpy/_core/__init__.py)\n","]\n","[autoreload of numpy failed: Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","             ^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/imp.py\", line 315, in reload\n","    return importlib.reload(module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/importlib/__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 621, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/__init__.py\", line 122, in <module>\n","    from ._core import (\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/_core/memmap.py\", line 5, in <module>\n","    from .numeric import uint8, ndarray, dtype\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/_core/numeric.py\", line 11, in <module>\n","    from . import multiarray\n","AttributeError: 'numpy.ufunc' object has no attribute '__module__'\n","]\n","[autoreload of numpy._core failed: Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"/usr/local/lib/python3.11/dist-packages/IPython/extensions/autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","             ^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/imp.py\", line 315, in reload\n","    return importlib.reload(module)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/lib/python3.11/importlib/__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 621, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/_core/__init__.py\", line 23, in <module>\n","    from . import multiarray\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/_core/multiarray.py\", line 97, in <module>\n","    _override___module__()\n","  File \"/usr/local/lib/python3.11/dist-packages/numpy/_core/multiarray.py\", line 93, in _override___module__\n","    ufunc.__module__ = \"numpy\"\n","    ^^^^^^^^^^^^^^^^\n","AttributeError: 'numpy.ufunc' object has no attribute '__module__'\n","]\n"]}],"source":["import sys\n","import os\n","\n","if SYS_PROJECT_DIR not in sys.path:\n","    sys.path.append(SYS_PROJECT_DIR)"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"3b2Ua0G9zm3G"},"outputs":[],"source":["os.chdir(SYS_SRC_DIR)\n","from dotenv import load_dotenv\n","_ = load_dotenv('.env')\n","from dotenv import dotenv_values\n","env_from_file = dotenv_values('.env')\n","env_from_file"]},{"cell_type":"markdown","metadata":{"id":"9XxM_22SI7-d"},"source":["# GPT-4 - Run"]},{"cell_type":"code","source":["!python run.py --model gpt-4o --vid_index 3"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"eUNw6QZ1vWTO","executionInfo":{"status":"ok","timestamp":1741572468327,"user_tz":-120,"elapsed":126956,"user":{"displayName":"Ni Ma","userId":"02197765216246653270"}},"outputId":"dead72a5-a5bc-4bbb-a142-eb5e9e06e2ab"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Directory already exists: ocr_results/gpt_results/gpt-4o\n","2025-03-10 02:06:01,812 ################################ Running gpt-4o Model on OCR Prompt ################################\n","\n","Processing videos:   0% 0/1 [00:00<?, ?video/s]Found existing scene collection: tt1sff1\n","\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:   0% 0/36 [00:00<?, ?scene/s]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:   3% 1/36 [00:03<02:11,  3.74s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:   6% 2/36 [00:05<01:36,  2.82s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:   8% 3/36 [00:08<01:33,  2.84s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:  11% 4/36 [00:12<01:42,  3.21s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:  14% 5/36 [00:16<01:47,  3.48s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:  17% 6/36 [00:20<01:47,  3.60s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:  19% 7/36 [00:23<01:44,  3.59s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:  22% 8/36 [00:27<01:37,  3.50s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:  25% 9/36 [00:32<01:50,  4.10s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:  28% 10/36 [00:39<02:04,  4.81s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:  31% 11/36 [00:47<02:27,  5.90s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:  33% 12/36 [00:50<02:00,  5.04s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:  36% 13/36 [00:54<01:51,  4.87s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:  39% 14/36 [00:58<01:40,  4.57s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:  42% 15/36 [01:02<01:31,  4.35s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:  44% 16/36 [01:05<01:17,  3.86s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:  47% 17/36 [01:08<01:09,  3.68s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:  50% 18/36 [01:11<01:01,  3.39s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:  53% 19/36 [01:14<00:53,  3.17s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:  56% 20/36 [01:16<00:47,  3.00s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:  58% 21/36 [01:21<00:51,  3.45s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:  61% 22/36 [01:27<01:00,  4.30s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:  64% 23/36 [01:30<00:51,  3.98s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:  67% 24/36 [01:33<00:43,  3.62s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:  69% 25/36 [01:37<00:41,  3.78s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:  72% 26/36 [01:40<00:35,  3.58s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:  75% 27/36 [01:44<00:32,  3.58s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:  78% 28/36 [01:46<00:25,  3.22s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:  81% 29/36 [01:49<00:20,  2.97s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:  83% 30/36 [01:51<00:16,  2.81s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:  86% 31/36 [01:54<00:13,  2.79s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:  89% 32/36 [01:57<00:11,  2.96s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:  92% 33/36 [02:00<00:08,  2.83s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:  94% 34/36 [02:01<00:04,  2.33s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6:  97% 35/36 [02:02<00:01,  1.99s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6: 100% 36/36 [02:03<00:00,  3.43s/scene]\n","2025-03-10 02:08:06,462 model results of m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6 saved to ocr_results/gpt_results/gpt-4o/ocr_2025-03-10_02-06-01/m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6_output.json\n","2025-03-10 02:08:06,477 results evaluations of m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6 saved to ocr_results/gpt_results/gpt-4o/ocr_2025-03-10_02-06-01/evaluations/m-z-0194c27c-aebe-75d0-812f-06fbeb60b7d6.json\n","Processing videos: 100% 1/1 [02:04<00:00, 124.66s/video]\n","Traceback (most recent call last):\n","  File \"/content/gdrive/MyDrive/Colab Notebooks/ocr/ocr/run.py\", line 175, in <module>\n","    main(args)\n","  File \"/content/gdrive/MyDrive/Colab Notebooks/ocr/ocr/run.py\", line 166, in main\n","    save_summary(current_run)\n","  File \"/content/gdrive/MyDrive/Colab Notebooks/ocr/ocr/utils.py\", line 146, in save_summary\n","    for model in os.listdir(os.path.join(\"ocr_results\",model_result)):\n","                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","NotADirectoryError: [Errno 20] Not a directory: 'ocr_results/summary_data_mistral-ocr-latest.json'\n"]}]},{"cell_type":"markdown","source":["# Mistral - Sample Run"],"metadata":{"id":"gu77vj8ricJf"}},{"cell_type":"code","source":["# image_url = \"https://storage.videodb.io/media/u-0e58db00-6548-4d3c-b8c1-4ddddc072787/m-z-0194c270-bbfb-7dd2-aaec-62d909b97b32/frames/frm-scn-tt1sff1-3.0-4.0-3.5/img-6f09a960-1078-4fb6-8b50-3edc6391aa56.png\""],"metadata":{"id":"yHeyavg4ick8"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["# import os\n","# from mistralai import Mistral\n","\n","# api_key = os.environ[\"MISTRAL_API_KEY\"]\n","# client = Mistral(api_key=api_key)\n","\n","# ocr_response = client.ocr.process(\n","#     model=\"mistral-ocr-latest\",\n","#     document={\n","#         \"type\": \"image_url\",\n","#         \"image_url\": image_url\n","#     }\n","# )"],"metadata":{"id":"PihH67P8ig9F"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["# ocr_response.pages[0].markdown"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"aB-VgqsfjjB9","executionInfo":{"status":"ok","timestamp":1741565096824,"user_tz":-120,"elapsed":25,"user":{"displayName":"Ni Ma","userId":"02197765216246653270"}},"outputId":"47fc537a-e0f4-4896-e15a-cc247e3b22fc"},"execution_count":null,"outputs":[{"output_type":"execute_result","data":{"text/plain":["[OCRPageObject(index=0, markdown='# HOW TO READ A CASE* \\n\\n## And Understand What it Means\\n\\n\"What is reading but silent conversation?\"\\n\\n- Walter Savage Landor\\n\\n[^0]\\n[^0]:    * Based in part on article by Julie Novkov at University at Albany, SUNY', images=[], dimensions=OCRPageDimensions(dpi=200, height=720, width=960))]"]},"metadata":{},"execution_count":51}]},{"cell_type":"markdown","source":["# Mistral - Run"],"metadata":{"id":"mKAVfB5nLFpr"}},{"cell_type":"code","source":["# !python run.py --model mistral --vid_index 4"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"MdOF5HE-LF_B","executionInfo":{"status":"ok","timestamp":1741567393281,"user_tz":-120,"elapsed":99645,"user":{"displayName":"Ni Ma","userId":"02197765216246653270"}},"outputId":"c39c1d95-b59c-45ac-db48-8584744c0216"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Directory already exists: ocr_results/mistral_results/mistral-ocr-latest\n","2025-03-10 00:41:54,255 ################################ Running mistral-ocr-latest Model on OCR Prompt ################################\n","\n","Processing videos:   0% 0/1 [00:00<?, ?video/s]Found existing scene collection: tt1sff1\n","\n","Processing scenes for video m-z-0194c27c-d107-7030-b990-0b5cc62f514a:   0% 0/29 [00:00<?, ?scene/s]\u001b[A\n","Processing scenes for video m-z-0194c27c-d107-7030-b990-0b5cc62f514a:   3% 1/29 [00:03<01:40,  3.58s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-d107-7030-b990-0b5cc62f514a:   7% 2/29 [00:06<01:27,  3.23s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-d107-7030-b990-0b5cc62f514a:  10% 3/29 [00:09<01:20,  3.12s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-d107-7030-b990-0b5cc62f514a:  14% 4/29 [00:12<01:18,  3.12s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-d107-7030-b990-0b5cc62f514a:  17% 5/29 [00:16<01:19,  3.30s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-d107-7030-b990-0b5cc62f514a:  21% 6/29 [00:19<01:13,  3.21s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-d107-7030-b990-0b5cc62f514a:  24% 7/29 [00:22<01:12,  3.27s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-d107-7030-b990-0b5cc62f514a:  28% 8/29 [00:26<01:09,  3.29s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-d107-7030-b990-0b5cc62f514a:  31% 9/29 [00:29<01:05,  3.28s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-d107-7030-b990-0b5cc62f514a:  34% 10/29 [00:32<01:02,  3.30s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-d107-7030-b990-0b5cc62f514a:  38% 11/29 [00:36<01:00,  3.33s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-d107-7030-b990-0b5cc62f514a:  41% 12/29 [00:39<00:54,  3.23s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-d107-7030-b990-0b5cc62f514a:  45% 13/29 [00:42<00:51,  3.22s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-d107-7030-b990-0b5cc62f514a:  48% 14/29 [00:45<00:49,  3.33s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-d107-7030-b990-0b5cc62f514a:  52% 15/29 [00:49<00:49,  3.50s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-d107-7030-b990-0b5cc62f514a:  55% 16/29 [00:53<00:45,  3.48s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-d107-7030-b990-0b5cc62f514a:  59% 17/29 [00:56<00:41,  3.48s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-d107-7030-b990-0b5cc62f514a:  62% 18/29 [00:59<00:37,  3.41s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-d107-7030-b990-0b5cc62f514a:  66% 19/29 [01:03<00:33,  3.35s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-d107-7030-b990-0b5cc62f514a:  69% 20/29 [01:06<00:29,  3.28s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-d107-7030-b990-0b5cc62f514a:  72% 21/29 [01:09<00:26,  3.37s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-d107-7030-b990-0b5cc62f514a:  76% 22/29 [01:13<00:23,  3.36s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-d107-7030-b990-0b5cc62f514a:  79% 23/29 [01:16<00:20,  3.36s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-d107-7030-b990-0b5cc62f514a:  83% 24/29 [01:19<00:16,  3.39s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-d107-7030-b990-0b5cc62f514a:  86% 25/29 [01:23<00:13,  3.47s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-d107-7030-b990-0b5cc62f514a:  90% 26/29 [01:26<00:10,  3.37s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-d107-7030-b990-0b5cc62f514a:  93% 27/29 [01:29<00:06,  3.25s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-d107-7030-b990-0b5cc62f514a:  97% 28/29 [01:32<00:03,  3.19s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-d107-7030-b990-0b5cc62f514a: 100% 29/29 [01:35<00:00,  3.31s/scene]\n","2025-03-10 00:43:31,160 model results of m-z-0194c27c-d107-7030-b990-0b5cc62f514a saved to ocr_results/mistral_results/mistral-ocr-latest/ocr_2025-03-10_00-41-53/m-z-0194c27c-d107-7030-b990-0b5cc62f514a_output.json\n","2025-03-10 00:43:31,408 results evaluations of m-z-0194c27c-d107-7030-b990-0b5cc62f514a saved to ocr_results/mistral_results/mistral-ocr-latest/ocr_2025-03-10_00-41-53/evaluations/m-z-0194c27c-d107-7030-b990-0b5cc62f514a.json\n","Processing videos: 100% 1/1 [01:37<00:00, 97.15s/video]\n"]}]},{"cell_type":"markdown","source":["# Gemini 2.0 Flash - Run"],"metadata":{"id":"z7J4iIM_bxVQ"}},{"cell_type":"code","source":["!python run.py --model gemini-2.0-flash --vid_index 5"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"BKq3WTvCb6VN","executionInfo":{"status":"ok","timestamp":1742173400281,"user_tz":-120,"elapsed":75312,"user":{"displayName":"Ni Ma","userId":"02197765216246653270"}},"outputId":"897fa1d6-199a-4acb-9534-6657f5b528b5"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Directory already exists: ocr_results/gemini_results/gemini-2.0-flash\n","2025-03-17 01:02:33,116 ################################ Running gemini-2.0-flash Model on OCR Prompt ################################\n","\n","Processing videos:   0% 0/1 [00:00<?, ?video/s]Found existing scene collection: tt1sff1\n","\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2:   0% 0/32 [00:00<?, ?scene/s]\u001b[A\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2:   3% 1/32 [00:02<01:04,  2.08s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2:   6% 2/32 [00:03<00:59,  1.98s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2:   9% 3/32 [00:05<00:55,  1.92s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2:  12% 4/32 [00:07<00:52,  1.87s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2:  16% 5/32 [00:09<00:49,  1.83s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2:  19% 6/32 [00:11<00:47,  1.84s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2:  22% 7/32 [00:13<00:47,  1.90s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2:  25% 8/32 [00:15<00:45,  1.88s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2:  28% 9/32 [00:16<00:43,  1.88s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2:  31% 10/32 [00:18<00:41,  1.90s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2:  34% 11/32 [00:20<00:39,  1.88s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2:  38% 12/32 [00:22<00:38,  1.94s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2:  41% 13/32 [00:24<00:37,  1.95s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2:  44% 14/32 [00:26<00:35,  1.97s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2:  47% 15/32 [00:28<00:33,  1.95s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2:  50% 16/32 [00:30<00:31,  1.95s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2:  53% 17/32 [00:32<00:29,  1.95s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2:  56% 18/32 [00:34<00:26,  1.92s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2:  59% 19/32 [00:36<00:25,  1.96s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2:  62% 20/32 [00:38<00:23,  1.97s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2:  66% 21/32 [00:40<00:21,  1.97s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2:  69% 22/32 [00:42<00:19,  1.99s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2:  72% 23/32 [00:46<00:22,  2.47s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2:  75% 24/32 [00:49<00:22,  2.84s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2:  78% 25/32 [00:53<00:21,  3.04s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2:  81% 26/32 [00:57<00:19,  3.23s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2:  84% 27/32 [01:00<00:16,  3.34s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2:  88% 28/32 [01:02<00:11,  2.94s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2:  91% 29/32 [01:04<00:07,  2.58s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2:  94% 30/32 [01:06<00:04,  2.30s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2:  97% 31/32 [01:07<00:02,  2.07s/scene]\u001b[A\n","Processing scenes for video m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2: 100% 32/32 [01:09<00:00,  2.18s/scene]\n","2025-03-17 01:03:44,099 model results of m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2 saved to ocr_results/gemini_results/gemini-2.0-flash/ocr_2025-03-17_01-02-32/m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2_output.json\n","2025-03-17 01:03:44,423 results evaluations of m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2 saved to ocr_results/gemini_results/gemini-2.0-flash/ocr_2025-03-17_01-02-32/evaluations/m-z-0194c27c-f30c-7803-b2ca-8f1026c940a2.json\n","Processing videos: 100% 1/1 [01:11<00:00, 71.30s/video]\n"]}]},{"cell_type":"markdown","metadata":{"id":"VIULUBYYr9wZ"},"source":["# Sleep"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"ydYCGxV_0RxJ"},"outputs":[],"source":["# import time\n","# time.sleep(30000)"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"e-1BoKn6_wzO"},"outputs":[],"source":[]}],"metadata":{"colab":{"machine_shape":"hm","provenance":[{"file_id":"1gUmEJOdcsEW_Ky0lQ0CfiMpMLlh9yZS8","timestamp":1727730998162},{"file_id":"1ZmxkTs9q1LEOjc_G7e2tHhSQrSYzdo6k","timestamp":1722897369093},{"file_id":"1hO-NEtS1mXUN5coJxBi7xi4ZaEwtY4nP","timestamp":1722896648514},{"file_id":"1Fg1tiO4POifUZ98Ap4sMy_YJI9edIFcB","timestamp":1718319079361},{"file_id":"11eqV3lWnKESuFMDCbj7MVxCkFPF5xB0M","timestamp":1711896356931},{"file_id":"13sKA0HqeV5lhwxl-oni5g3ZvX3NQTVm5","timestamp":1707337821789},{"file_id":"1IG-FnlrJPPk0Y8VoE84YnJ0Iab76pwKZ","timestamp":1696459542919},{"file_id":"1IXzuTWOlQ7Eb9BaMz0WziXedDD0aPVGx","timestamp":1696249018299},{"file_id":"1VbnFSEkWPWt6TOHG2UA-vM2S6SakpPuv","timestamp":1696170946196},{"file_id":"1fJk_HogsfbBUTmi6HDfvgG6vDyAAYLYI","timestamp":1696154984193},{"file_id":"1dk7XJxs88yBz31vI5xdGV7oWFoGDFYT1","timestamp":1696021208118},{"file_id":"1TBZaaR2MthzbXRpsEe2dzvg8IV23jGoP","timestamp":1695852949509},{"file_id":"1t5l8tafcqZy9zpufyiEjls1tWvwPKTo5","timestamp":1695797899519},{"file_id":"1y2rJsf0XpxJrSH3ehZtsmAuUWpyEAERn","timestamp":1695319794549},{"file_id":"1L4_O-2GVBPWurs2Cvqjo6v9XBGAwt_Sz","timestamp":1695112367549},{"file_id":"1TlICrM-qRriJgHwYSItoc9HQKwvPwnTo","timestamp":1694436436514},{"file_id":"1iRGQ8w5C7LGrWIMlkakUb8lc9eO0SiBc","timestamp":1693948173151},{"file_id":"1sc-2EElLa9Jf8USIeIaW09vml5u-Bdqq","timestamp":1693348861729}],"toc_visible":true,"collapsed_sections":["gu77vj8ricJf"],"mount_file_id":"11eqV3lWnKESuFMDCbj7MVxCkFPF5xB0M","authorship_tag":"ABX9TyOZV/3paPFgO8sfH4p4T4MP"},"kernelspec":{"display_name":"Python 3","name":"python3"},"language_info":{"name":"python"}},"nbformat":4,"nbformat_minor":0}